### ISSUE TITLE
Wrong inference for `str(const)`

### ISSUE THREAD

**Issue by jkmnt** [pylint-dev/astroid#2994]
These return `Const("")`, which is really misleading (even an `Uninferable` is better):

```python
code = """
    s = str("abc") #@
    s = str(1) #@
    s = str(None) #@
    s = str(2 + 2) #@
    """
```

Looking at `brain_builtin_inference.py`, the `def infer_str()` function is hardcoded to always return `Const("")`.
Next to it is the `def infer_int()`, which at least tries to examine `int()` arguments.

I'll submit a small PR fixing it at least for const argument.

**Comment by Pierre-Sassoulas** [comment]
pylint is tested well enough automatically that an agentic AI could be remarquably efficient fixing this in pylint. But it would still affect plugins. Sounds like an issue that need investigation. (Maybe a new pylint-speficic checker/plugin to suggest catching inference error explicitely when inferreing + deprecation would make this less painful? There was some exploration on inference error catching earlier but using the generic safe_infer wasn't flagged as bad pragtice at the time)
